lib/spack/spack/schema/config.py

```diff
@@ -41,6 +41,8 @@
             'additionalProperties': False,
             'properties': {
                 'install_tree': {'type': 'string'},
+                'install_hash_length': {'type': 'integer', 'minimum': 0},
```

Comment:
Can you set this to be "1" instead of "0"? I think creating a distinction between 0 and >0 is worthwhile. Not using a hash at all increases the collision likelihood to the point where IMO spack would not be usable. Certainly a hash length of 1 would also be pretty bad but forcing the use of __some__ hash forces the user to consider what the length should be.